src/h2/stream.py

```diff
@@ -315,21 +307,23 @@ def recv_push_promise(self, previous_state: StreamState) -> list[Event]:
         event.parent_stream_id = self.stream_id
         return [event]
 
-    def send_end_stream(self, previous_state: StreamState) -> None:
+    def send_end_stream(self, previous_state: StreamState) -> list[Event]:
         """
         Called when an attempt is made to send END_STREAM in the
         HALF_CLOSED_REMOTE state.
         """
         self.stream_closed_by = StreamClosedBy.SEND_END_STREAM
+        return []
```

Comment:
What's the idea with this API change? To make it look uniform with other similar methods?
I'm not sure if this will actually make it easier to use, if we keep sprinkling this pattern around and downstream users have to keep checking the returned list for potential events even though it should be explicitly clear that no events can happen.

Please explicitly document in changelog & `versionadded` comment.

Reply:
Ah - this is probably PR-bleed-over from https://github.com/python-hyper/h2/pull/1297.
Generally it now makes more sense - thanks!

If this only affects internal methods from the state machine, I think we can skip the `versionadded` comments. WDYT?

Reply:
> If this only affects internal methods from the state machine, I think we can skip the versionadded comments. WDYT?

Agreed.